src/keria/utils/openapi.py

```diff
@@ -0,0 +1,385 @@
+                        return value
+                        
+                    def __init__(self, **kwargs):
+                        super().__init__(**kwargs)
+                        self.metadata['oneOf'] = [
+                            {'type': 'string'},
+                            {'type': 'array', 'items': {'type': 'string'}}
+                        ]
+                
+                if isOptional:
+                    marshmallowField = StringOrStringArrayField(
+                        required=False,
+                        missing=mm_fields.missing_,
+                        default=mm_fields.missing_,
+                        allow_none=False
+                    )
+                    fieldDef = (key, Optional[customType], field(default=None))
+                else:
+                    marshmallowField = StringOrStringArrayField(
+                        required=True,
+                        missing=mm_fields.missing_,
+                        default=mm_fields.missing_
+                    )
+                    fieldDef = (key, customType, field())
+                return fieldDef, marshmallowField
+        
+        # For other Union types, fall back to Raw field with anyOf
+        class UnionField(mm_fields.Field):
+            def _serialize(self, value, attr, obj, **kwargs):
+                return value
+                
+            def _deserialize(self, value, attr, data, **kwargs):
+                return value
+                
+            def __init__(self, union_type, **kwargs):
+                super().__init__(**kwargs)
+                self.metadata['anyOf'] = []
+                for arg in union_type.__args__:
+                    if arg is str:
+                        self.metadata['anyOf'].append({'type': 'string'})
+                    elif hasattr(arg, '__origin__') and arg.__origin__ is list:
+                        if arg.__args__ and arg.__args__[0] is str:
+                            self.metadata['anyOf'].append({'type': 'array', 'items': {'type': 'string'}})
+                        else:
+                            self.metadata['anyOf'].append({'type': 'array'})
+        
+        if isOptional:
+            marshmallowField = UnionField(
+                customType,
+                required=False,
+                missing=mm_fields.missing_,
+                default=mm_fields.missing_,
+                allow_none=False
+            )
+            fieldDef = (key, Optional[customType], field(default=None))
+        else:
+            marshmallowField = UnionField(
+                customType,
+                required=True,
+                missing=mm_fields.missing_,
+                default=mm_fields.missing_
+            )
+            fieldDef = (key, customType, field())
+        return fieldDef, marshmallowField
+    
+    # Check if this is a generic type like List[SomeClass]
+    elif hasattr(customType, '__origin__'):
+        if customType.__origin__ is list:
```

Comment:
Again, on code style, we should probably pull out this logic branch into an appropriately named function. I like to capture unlabeled abstractions, such as branches of an if else chain, with function names to make the intent clear. This is likely the best approach to the below logic branch.

Extract it and name the inner function.